Library/Homebrew/tap_constants.rb

```diff
@@ -1,6 +1,8 @@
 # typed: strict
 # frozen_string_literal: true
 
+HOMEBREW_TAP_DIRECTORY = T.let((HOMEBREW_LIBRARY/"Taps").freeze, Pathname)
```

Comment:
This could potentially also go in `config.rb` instead. I have no real preference between the two.

Reply:
`config.rb` seems better. `config.rb` has directory constants while this file is all regexes.

Reply:
Moving it breaks the test requires somehow. I'll look more into it later. I didn't expect that.

Reply:
Integration tests use their own `config.rb`

Reply:
Addressed in https://github.com/Homebrew/brew/pull/18010/commits/1fe16a5e35952326a97a665ffe9ddde18ee7978b